### PR TITLE
Refactor/lecture page1

### PR DESCRIPTION
**Pull request by drouba**

### CODE REVIEW

File: modules/ensure_logged_in.php [hunk 1]
```diff
@@ -2,4 +2,11 @@
...
 require_once(dirname(__FILE__)."/../common.php");
 
-ensure_logged_in();
+if(!isset($in_settings)){
+    $in_settings = false;
+}
+
+if(!$in_settings){
+    ensure_logged_in();
+}
```
Comment (jamestjw) [inline modules/ensure_logged_in.php:11]:
```php
if(isset($in_settings) ? $in_settings : false){
     ensure_logged_in();
 }
 ```
 
 you could do this, its cleaner

File: public/login.php [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
     if ($is_success == 0) {
         $_SESSION["alert_message"] = "Invalid email or Password!";
+        echo $_SESSION["alert_message"];
```
Comment (jamestjw) [inline public/login.php:29]: Im guessing the the echo here wasn't intentional?

File: public/view_lecture_teams.php [hunk 3]
```diff
@@ -72,11 +72,58 @@
...
     <br><a href="create_team.php?id=<?php echo $lecture_page_id ?>">Create new teams</a> 
-<?php } ?>
\ No newline at end of file
+<?php }else {
+    try {
+        $all_student_team = TeamMember::includes(['teams' => 'lectures'])->where(array('user_id' => get_users_id(),));
+    
+    } catch (PDOException $error) {
+        echo "<br>" . $error->getMessage();
+    }
```
Comment (jamestjw) [inline public/view_lecture_teams.php:86]: hmm ok i have a différent proposal to find the team that the student is in,
```php
$student_team = Team::joins(["team_members"])->where(["user_id"=>get_users_id(), "lecture_id"=>$lecture_page_id]))
```

Comment (jamestjw) [reply public/view_lecture_teams.php:86]: its a lot simpler and easier to understand too

**Review by jamestjw — CHANGES_REQUESTED** [review]

File: modules/ensure_logged_in.php (hunk 1 above)
Comment (drouba) [reply modules/ensure_logged_in.php:11]: done

Comment (drouba) [reply modules/ensure_logged_in.php:11]: it doesn't work

File: public/login.php (hunk 2 above)
Comment (drouba) [reply public/login.php:29]: yup removed

File: public/view_lecture_teams.php (hunk 3 above)
Comment (drouba) [reply public/view_lecture_teams.php]: doesn't work creates an error

Comment (drouba) [reply public/view_lecture_teams.php]: done

File: modules/ensure_logged_in.php (hunk 1 above)
Comment (jamestjw) [reply modules/ensure_logged_in.php:11]: @drouba  in what sense?

**Review by jamestjw — APPROVED** [review]

**Pull request merged**